vm/actor/src/builtin/market/mod.rs

```diff
@@ -4,7 +4,687 @@
+        rt: &mut RT,
+        params: VerifyDealsOnSectorProveCommitParams,
+    ) -> Result<DealWeight, ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_type(std::iter::once(&*MINER_ACTOR_CODE_ID))?;
+        let miner_addr = rt.message().from().clone();
+        let mut total_deal_space_time = BigInt::zero();
+        let mut deal_weight = BigInt::zero();
+
+        rt.transaction::<State, Result<(), ActorError>, _>(|st, rt| {
+            // if there are no dealIDs, it is a CommittedCapacity sector
+            // and the totalDealSpaceTime should be zero
+            let mut states = Amt::load(&st.states, rt.store())
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?;
+            let proposals = Amt::load(&st.proposals, rt.store())
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?;
+
+            for id in &params.deal_ids {
+                let mut deal: DealState = states
+                    .get(*id)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?
+                    .ok_or_else(|| {
+                        ActorError::new(ExitCode::ErrIllegalState, "Get deal error".to_owned())
+                    })?;
+                let proposal: DealProposal = proposals
+                    .get(*id)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?
+                    .ok_or_else(|| {
+                        ActorError::new(ExitCode::ErrIllegalState, "Get deal error".to_owned())
+                    })?;
+
+                validate_deal_can_activate(
+                    rt.curr_epoch(),
+                    &miner_addr,
+                    params.sector_expiry,
+                    &proposal,
+                )?;
+
+                deal.sector_start_epoch = OptionalEpoch(Some(rt.curr_epoch()));
+                states
+                    .set(*id, deal)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+                // compute deal weight
+                let deal_space_time = proposal.duration() * proposal.piece_size.0;
+                total_deal_space_time += deal_space_time;
+            }
+            st.states = states
+                .flush()
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            let epoch_value = params
+                .sector_expiry
+                .checked_sub(rt.curr_epoch())
+                .ok_or_else(|| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalArgument,
+                        format!(
+                            "invalid sector expiry epoch {:?} ahead of current {:?}",
+                            params.sector_expiry,
+                            rt.curr_epoch()
+                        ),
+                    )
+                })?;
```

Comment:
I know this may sound weird, but you actually need to do 
```
.checked_sub(..).ok_or(1u8)
```
lol

I thought through the logic and their impl doesn't error when this value is negative, so this needs to succeed, divide 0 by a valid number and return successfully and/or panic on a divide by 0

https://github.com/filecoin-project/specs-actors/issues/280